core/chains/evm/logpoller/log_poller.go

```diff
@@ -529,22 +516,23 @@ func (lp *logPoller) run() {
 			}
 			lp.PollAndSaveLogs(lp.ctx, start)
 		case <-backupLogPollTick:
+			if lp.backupPollerBlockDelay == 0 {
+				continue // backup poller is disabled
+			}
 			// Backup log poller:  this serves as an emergency backup to protect against eventual-consistency behavior
 			// of an rpc node (seen occasionally on optimism, but possibly could happen on other chains?).  If the first
 			// time we request a block, no logs or incomplete logs come back, this ensures that every log is eventually
-			// re-requested after it is finalized.  This doesn't add much overhead, because we can request all of them
-			// in one shot, since we don't need to worry about re-orgs after finality depth, and it runs 100x less
-			// frequently than the primary log poller.
-
-			// If pollPeriod is set to 1 block time, backup log poller will run once every 100 blocks
-			const backupPollerBlockDelay = 100
+			// re-requested after it is finalized. This doesn't add much overhead, because we can request all of them
+			// in one shot, since we don't need to worry about re-orgs after finality depth, and it runs far less
+			// frequently than the primary log poller (instead of roughly once per block it runs once roughly once every
+			// lp.backupPollerDelay blocks--with default settings about 100x less frequently).
 
-			backupLogPollTick = time.After(utils.WithJitter(backupPollerBlockDelay * lp.pollPeriod))
+			backupLogPollTick = time.After(utils.WithJitter(time.Duration(lp.backupPollerBlockDelay) * lp.pollPeriod))
```

Comment:
since `backupPollerBlockDelay` is an `int64` would it make sense to add time unit here as well? e.g.`time.Duration(lp.backupPollerBlockDelay * time.Second)` so that it's more explicit what's the actual interval?

Reply:
No, this would give the wrong answer because `backupPollerBlockDelay` is the number of blocks, not a time duration.

I was also surprised to find out how misleading the syntax is for multiplying a duration by an integer is in golang. It's really ugly:
https://stackoverflow.com/questions/17573190/how-to-multiply-duration-by-integer

Basically, you're required to convert the integer into a time.Duration even though it's not a time interval.

Because the `time.Second` constant is defined as 1000000000 rather than 1, adding `* time.Second` there would give an answer that is a billion times too large (waiting eg 100-billion blocks instead of the default 100 blocks)!